### ISSUE TITLE
IndexOutofBounds with Google maps

### ISSUE THREAD

**Issue by Lineflyer** [cgeo/cgeo#3507]
2014.01.03-NB1:

I was using Google maps today (instead of the OSM-offline I normally use).
I came across several exceptions like these while using the live map for some minutes.
The live map keeps working but these exceptions can be seen in the log:
```
13:24:02.234 Error cgeo 24583  GoogleMapView.draw
13:24:02.234 Error cgeo 24583  java.lang.ArrayIndexOutOfBoundsException: length=0; index=0
13:24:02.234 Error cgeo 24583  	at com.google.android.maps.ItemizedOverlay.getIndexToDraw(ItemizedOverlay.java:211)
13:24:02.234 Error cgeo 24583  	at com.google.android.maps.ItemizedOverlay.draw(ItemizedOverlay.java:240)
13:24:02.234 Error cgeo 24583  	at cgeo.geocaching.maps.google.GoogleCacheOverlay.superDraw(GoogleCacheOverlay.java:101)
13:24:02.234 Error cgeo 24583  	at cgeo.geocaching.maps.AbstractItemizedOverlay.draw(AbstractItemizedOverlay.java:44)
13:24:02.234 Error cgeo 24583  	at cgeo.geocaching.maps.CachesOverlay.draw(CachesOverlay.java:104)
13:24:02.234 Error cgeo 24583  	at cgeo.geocaching.maps.google.GoogleCacheOverlay.draw(GoogleCacheOverlay.java:66)
13:24:02.234 Error cgeo 24583  	at com.google.android.maps.Overlay.draw(Overlay.java:179)
13:24:02.234 Error cgeo 24583  	at com.google.android.maps.OverlayBundle.draw(OverlayBundle.java:42)
13:24:02.234 Error cgeo 24583  	at com.google.android.maps.MapView.onDraw(MapView.java:532)
13:24:02.234 Error cgeo 24583  	at android.view.View.draw(View.java:14853)
13:24:02.234 Error cgeo 24583  	at cgeo.geocaching.maps.google.GoogleMapView.draw(GoogleMapView.java:63)
13:24:02.234 Error cgeo 24583  	at android.view.View.draw(View.java:14734)
13:24:02.234 Error cgeo 24583  	at android.view.ViewGroup.drawChild(ViewGroup.java:3316)
13:24:02.234 Error cgeo 24583  	at android.view.ViewGroup.dispatchDraw(ViewGroup.java:3153)
13:24:02.234 Error cgeo 24583  	at android.view.View.draw(View.java:14732)
13:24:02.234 Error cgeo 24583  	at android.view.ViewGroup.drawChild(ViewGroup.java:3316)
13:24:02.234 Error cgeo 24583  	at android.view.ViewGroup.dispatchDraw(ViewGroup.java:3153)
13:24:02.234 Error cgeo 24583  	at android.view.View.draw(View.java:14732)
13:24:02.234 Error cgeo 24583  	at android.view.ViewGroup.drawChild(ViewGroup.java:3316)
13:24:02.234 Error cgeo 24583  	at android.view.ViewGroup.dispatchDraw(ViewGroup.java:3153)
13:24:02.234 Error cgeo 24583  	at android.view.View.draw(View.java:14732)
13:24:02.234 Error cgeo 24583  	at android.view.ViewGroup.drawChild(ViewGroup.java:3316)
13:24:02.234 Error cgeo 24583  	at android.view.ViewGroup.dispatchDraw(ViewGroup.java:3153)
13:24:02.234 Error cgeo 24583  	at android.view.View.draw(View.java:14732)
13:24:02.234 Error cgeo 24583  	at android.view.ViewGroup.drawChild(ViewGroup.java:3316)
13:24:02.234 Error cgeo 24583  	at android.view.ViewGroup.dispatchDraw(ViewGroup.java:3153)
13:24:02.234 Error cgeo 24583  	at android.view.View.draw(View.java:14732)
13:24:02.234 Error cgeo 24583  	at android.view.ViewGroup.drawChild(ViewGroup.java:3316)
13:24:02.234 Error cgeo 24583  	at android.view.ViewGroup.dispatchDraw(ViewGroup.java:3153)
13:24:02.234 Error cgeo 24583  	at android.view.View.draw(View.java:14732)
13:24:02.234 Error cgeo 24583  	at android.view.ViewGroup.drawChild(ViewGroup.java:3316)
13:24:02.234 Error cgeo 24583  	at android.view.ViewGroup.dispatchDraw(ViewGroup.java:3153)
13:24:02.234 Error cgeo 24583  	at android.sec.multiwindow.impl.BorderFrameLayout.dispatchDraw(BorderFrameLayout.java:46)
13:24:02.234 Error cgeo 24583  	at android.view.View.draw(View.java:14732)
13:24:02.234 Error cgeo 24583  	at android.view.ViewGroup.drawChild(ViewGroup.java:3316)
13:24:02.234 Error cgeo 24583  	at android.view.ViewGroup.dispatchDraw(ViewGroup.java:3153)
13:24:02.234 Error cgeo 24583  	at android.view.View.draw(View.java:14856)
13:24:02.234 Error cgeo 24583  	at android.widget.FrameLayout.draw(FrameLayout.java:467)
13:24:02.234 Error cgeo 24583  	at com.android.internal.policy.impl.PhoneWindow$DecorView.draw(PhoneWindow.java:2621)
13:24:02.234 Error cgeo 24583  	at android.view.ViewRootImpl.drawSoftware(ViewRootImpl.java:2787)
13:24:02.234 Error cgeo 24583  	at android.view.ViewRootImpl.draw(ViewRootImpl.java:2700)
13:24:02.234 Error cgeo 24583  	at android.view.ViewRootImpl.performDraw(ViewRootImpl.java:2538)
13:24:02.234 Error cgeo 24583  	at android.view.ViewRootImpl.performTraversals(ViewRootImpl.java:2154)
13:24:02.234 Error cgeo 24583  	at android.view.ViewRootImpl.doTraversal(ViewRootImpl.java:1249)
13:24:02.234 Error cgeo 24583  	at android.view.ViewRootImpl$TraversalRunnable.run(ViewRootImpl.java:6364)
13:24:02.234 Error cgeo 24583  	at android.view.Choreographer$CallbackRecord.run(Choreographer.java:791)
13:24:02.234 Error cgeo 24583  	at android.view.Choreographer.doCallbacks(Choreographer.java:591)
13:24:02.234 Error cgeo 24583  	at android.view.Choreographer.doFrame(Choreographer.java:561)
13:24:02.234 Error cgeo 24583  	at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:777)
13:24:02.234 Error cgeo 24583  	at android.os.Handler.handleCallback(Handler.java:730)
13:24:02.234 Error cgeo 24583  	at android.os.Handler.dispatchMessage(Handler.java:92)
13:24:02.234 Error cgeo 24583  	at android.os.Looper.loop(Looper.java:137)
13:24:02.234 Error cgeo 24583  	at android.app.ActivityThread.main(ActivityThread.java:5419)
13:24:02.234 Error cgeo 24583  	at java.lang.reflect.Method.invokeNative(Native Method)
13:24:02.234 Error cgeo 24583  	at java.lang.reflect.Method.invoke(Method.java:525)
13:24:02.234 Error cgeo 24583  	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1209)
13:24:02.234 Error cgeo 24583  	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1025)
13:24:02.234 Error cgeo 24583  	at dalvik.system.NativeStart.main(Native Method)
```

Another example:
```


14:06:45.765 Error cgeo 29968  GoogleMapView.draw
14:06:45.765 Error cgeo 29968  java.lang.ArrayIndexOutOfBoundsException: length=444; index=444
14:06:45.765 Error cgeo 29968  	at com.google.android.maps.ItemizedOverlay.getIndexToDraw(ItemizedOverlay.java:211)
14:06:45.765 Error cgeo 29968  	at com.google.android.maps.ItemizedOverlay.draw(ItemizedOverlay.java:240)
14:06:45.765 Error cgeo 29968  	at cgeo.geocaching.maps.google.GoogleCacheOverlay.superDraw(GoogleCacheOverlay.java:101)
14:06:45.765 Error cgeo 29968  	at cgeo.geocaching.maps.AbstractItemizedOverlay.draw(AbstractItemizedOverlay.java:44)
14:06:45.765 Error cgeo 29968  	at cgeo.geocaching.maps.CachesOverlay.draw(CachesOverlay.java:104)
14:06:45.765 Error cgeo 29968  	at cgeo.geocaching.maps.google.GoogleCacheOverlay.draw(GoogleCacheOverlay.java:66)
14:06:45.765 Error cgeo 29968  	at com.google.android.maps.Overlay.draw(Overlay.java:179)
14:06:45.765 Error cgeo 29968  	at com.google.android.maps.OverlayBundle.draw(OverlayBundle.java:45)
14:06:45.765 Error cgeo 29968  	at com.google.android.maps.MapView.onDraw(MapView.java:532)
14:06:45.765 Error cgeo 29968  	at android.view.View.draw(View.java:14853)
14:06:45.765 Error cgeo 29968  	at cgeo.geocaching.maps.google.GoogleMapView.draw(GoogleMapView.java:63)
14:06:45.765 Error cgeo 29968  	at android.view.View.draw(View.java:14734)
14:06:45.765 Error cgeo 29968  	at android.view.ViewGroup.drawChild(ViewGroup.java:3316)
14:06:45.765 Error cgeo 29968  	at android.view.ViewGroup.dispatchDraw(ViewGroup.java:3153)
14:06:45.765 Error cgeo 29968  	at android.view.View.draw(View.java:14732)
14:06:45.765 Error cgeo 29968  	at android.view.ViewGroup.drawChild(ViewGroup.java:3316)
14:06:45.765 Error cgeo 29968  	at android.view.ViewGroup.dispatchDraw(ViewGroup.java:3153)
14:06:45.765 Error cgeo 29968  	at android.view.View.draw(View.java:14732)
14:06:45.765 Error cgeo 29968  	at android.view.ViewGroup.drawChild(ViewGroup.java:3316)
14:06:45.765 Error cgeo 29968  	at android.view.ViewGroup.dispatchDraw(ViewGroup.java:3153)
14:06:45.765 Error cgeo 29968  	at android.view.View.draw(View.java:14732)
14:06:45.765 Error cgeo 29968  	at android.view.ViewGroup.drawChild(ViewGroup.java:3316)
14:06:45.765 Error cgeo 29968  	at android.view.ViewGroup.dispatchDraw(ViewGroup.java:3153)
14:06:45.765 Error cgeo 29968  	at android.view.View.draw(View.java:14732)
14:06:45.765 Error cgeo 29968  	at android.view.ViewGroup.drawChild(ViewGroup.java:3316)
14:06:45.765 Error cgeo 29968  	at android.view.ViewGroup.dispatchDraw(ViewGroup.java:3153)
14:06:45.765 Error cgeo 29968  	at android.view.View.draw(View.java:14732)
14:06:45.765 Error cgeo 29968  	at android.view.ViewGroup.drawChild(ViewGroup.java:3316)
14:06:45.765 Error cgeo 29968  	at android.view.ViewGroup.dispatchDraw(ViewGroup.java:3153)
14:06:45.765 Error cgeo 29968  	at android.view.View.draw(View.java:14732)
14:06:45.765 Error cgeo 29968  	at android.view.ViewGroup.drawChild(ViewGroup.java:3316)
14:06:45.765 Error cgeo 29968  	at android.view.ViewGroup.dispatchDraw(ViewGroup.java:3153)
14:06:45.765 Error cgeo 29968  	at android.sec.multiwindow.impl.BorderFrameLayout.dispatchDraw(BorderFrameLayout.java:46)
14:06:45.765 Error cgeo 29968  	at android.view.View.draw(View.java:14732)
14:06:45.765 Error cgeo 29968  	at android.view.ViewGroup.drawChild(ViewGroup.java:3316)
14:06:45.765 Error cgeo 29968  	at android.view.ViewGroup.dispatchDraw(ViewGroup.java:3153)
14:06:45.765 Error cgeo 29968  	at android.view.View.draw(View.java:14856)
14:06:45.765 Error cgeo 29968  	at android.widget.FrameLayout.draw(FrameLayout.java:467)
14:06:45.765 Error cgeo 29968  	at com.android.internal.policy.impl.PhoneWindow$DecorView.draw(PhoneWindow.java:2621)
14:06:45.765 Error cgeo 29968  	at android.view.ViewRootImpl.drawSoftware(ViewRootImpl.java:2787)
14:06:45.765 Error cgeo 29968  	at android.view.ViewRootImpl.draw(ViewRootImpl.java:2700)
14:06:45.765 Error cgeo 29968  	at android.view.ViewRootImpl.performDraw(ViewRootImpl.java:2538)
14:06:45.765 Error cgeo 29968  	at android.view.ViewRootImpl.performTraversals(ViewRootImpl.java:2154)
14:06:45.765 Error cgeo 29968  	at android.view.ViewRootImpl.doTraversal(ViewRootImpl.java:1249)
14:06:45.765 Error cgeo 29968  	at android.view.ViewRootImpl$TraversalRunnable.run(ViewRootImpl.java:6364)
14:06:45.765 Error cgeo 29968  	at android.view.Choreographer$CallbackRecord.run(Choreographer.java:791)
14:06:45.765 Error cgeo 29968  	at android.view.Choreographer.doCallbacks(Choreographer.java:591)
14:06:45.765 Error cgeo 29968  	at android.view.Choreographer.doFrame(Choreographer.java:561)
14:06:45.765 Error cgeo 29968  	at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:777)
14:06:45.765 Error cgeo 29968  	at android.os.Handler.handleCallback(Handler.java:730)
14:06:45.765 Error cgeo 29968  	at android.os.Handler.dispatchMessage(Handler.java:92)
14:06:45.765 Error cgeo 29968  	at android.os.Looper.loop(Looper.java:137)
14:06:45.765 Error cgeo 29968  	at android.app.ActivityThread.main(ActivityThread.java:5419)
14:06:45.765 Error cgeo 29968  	at java.lang.reflect.Method.invokeNative(Native Method)
14:06:45.765 Error cgeo 29968  	at java.lang.reflect.Method.invoke(Method.java:525)
14:06:45.765 Error cgeo 29968  	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1209)
14:06:45.765 Error cgeo 29968  	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1025)
14:06:45.765 Error cgeo 29968  	at dalvik.system.NativeStart.main(Native Method)
```

**Comment by Lineflyer** [comment]
Posted an updated issue in #4611 

(Found this only after creating the new issue with some more information in it).
